libraries/AC_AutoTune/AC_AutoTune.cpp

```diff
@@ -1133,102 +1160,120 @@ void AC_AutoTune::load_gains(enum GainType gain_type)
     }
 }
 
+// save or discard gains
 // save_tuning_gains - save the final tuned gains for each axis
 // save discovered gains to eeprom if autotuner is enabled (i.e. switch is in the high position)
-void AC_AutoTune::save_tuning_gains()
+void AC_AutoTune::save_tuning_gains(bool save)
 {
     // see if we successfully completed tuning of at least one axis
     if (axes_completed == 0) {
+        if (!save) {
```

Comment:
Why don't we want to reset here?

Reply:
This should preserve the existing behavior, if you disarm without completing you can re-arm and carry on.

Reply:
That's definitely worth a comment.

Reply:
I have added a comment, to be honest that is just to preserve existing behavior, I'm not sure why we cant skip the reset in all cases, seems logical that it would continue if it had not finished, no matter the mode disarmed in.